library/core/src/ffi/c_str.rs

```diff
@@ -875,4 +875,42 @@ mod verify {
             assert!(c_str.is_safe());
         }
     }
+
+    // pub const fn to_bytes(&self) -> &[u8]
+    #[kani::proof]
+    #[kani::unwind(32)]
+    fn check_to_bytes() {
+        const MAX_SIZE: usize = 32;
+        let string: [u8; MAX_SIZE] = kani::any();
+        let slice = kani::slice::any_slice_of_array(&string);
+
+        let result = CStr::from_bytes_until_nul(slice);
+        if let Ok(c_str) = result {
+            // Find the index of the first null byte in the slice since
+            // from_bytes_until_nul stops by there
+            let end_idx = slice.iter().position(|x| *x == 0).unwrap();
```

Comment:
Can you do something like this instead:
```rust
    let bytes = c_str.to_bytes();
    let end_idx = bytes.len();
    assert_eq!(bytes, &slice[..end_idx]);
```
BTW, the check above is a nice to have, but not required since we are focused on safety verification. You could also use `#[ensures()]` to check the result.

Reply:
In this case, I think the harness should be able to access the slice, but I encapsulated the boiler plate code.

Reply:
A workaround is let arbitrary_cstr also return the input slice.

Reply:
Fixed.